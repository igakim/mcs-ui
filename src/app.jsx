import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import App from './view/Home';

export default (store) => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('root'),
  );
};
