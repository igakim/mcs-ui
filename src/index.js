import '@babel/polyfill';

// Import Components:
import Button from './components/button/Button';
import Menu from './components/menu/Menu';
import Section from './components/section/Section';
import Card from './components/card/Card';
import Form from './components/form/Form';

export {
  Button,
  Menu,
  Section,
  Card,
  Form,
};
