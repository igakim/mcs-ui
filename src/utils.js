export const isEqualTypes = (type, compareType) => (
  type && type === compareType
);

export const getButtonType = type => (
  type && `mcs-button-${type}`
);

export const getClassNameFromArray = classesList => classesList
  .filter(el => (el !== undefined && el !== null))
  .join('-');

export const isClassComponent = component => (!!(
  typeof component === 'function'
    && component.prototype
    && component.prototype.isReactComponent
));

export const isFunctionComponent = component => (!!((
  typeof component === 'function'
    && !(
      component.prototype
      && component.prototype.isReactComponent
    )
)));

export const isReactComponent = component => (!!((
  isClassComponent(component)
    || isFunctionComponent(component)
)));
