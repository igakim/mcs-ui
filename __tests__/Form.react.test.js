import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Form } from '../src';
import { Button } from '../src';

configure({ adapter: new Adapter() });

test('component should match snapshot', () => {
  const { Input } = Form;

  const component = renderer.create(
    <Form>
      <Input type="text" name="login" placeholder="Enter your name" />
      <Input type="password" name="password" placeholder="Your password" />
      <Button type="primary" htmlType="submit">Log in</Button>
    </Form>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should match snapshot', () => {
  const { SearchInput } = Form;

  const component = renderer.create(
    <Form>
      <SearchInput type="text" name="login" buttonText="Find" placeholder="Enter your name" />
    </Form>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should match snapshot', () => {
  const { SearchInput } = Form;

  const component = renderer.create(
    <Form>
      <SearchInput type="text" name="login" buttonText="Find" placeholder="Enter your name" prefix="O" />
    </Form>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Form /> component should work properly', () => {
  const { Input } = Form;

  const wrapper = shallow(
    <Form>
      <Input type="text" name="login" placeholder="Enter your name" />
      <Button type="primary" htmlType="submit">Log in</Button>
    </Form>
  );

  const input = wrapper.find(Input).simulate('change', {target: {value: 'abc', name: 'login'}});

  expect(wrapper.state().fields.login.value).toBe('abc');

  const button = wrapper.find(Button).shallow().simulate('click');

});
