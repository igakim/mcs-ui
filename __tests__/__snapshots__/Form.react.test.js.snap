// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should match snapshot 1`] = `
<form
  className="mcs-form "
  onSubmit={[Function]}
>
  <input
    className="mcs-form-input "
    name="login"
    onChange={[Function]}
    placeholder="Enter your name"
    type="text"
    value=""
  />
  <input
    className="mcs-form-input "
    name="password"
    onChange={[Function]}
    placeholder="Your password"
    type="password"
  />
  <button
    className="mcs-button mcs-button-primary"
    disabled={false}
    onChange={[Function]}
    type="submit"
  >
    Log in
  </button>
</form>
`;

exports[`component should match snapshot 2`] = `
<form
  className="mcs-form "
  onSubmit={[Function]}
>
  <div
    className="mcs-form-search "
  >
    <input
      className="mcs-form-search-input"
      name="login"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <button
      className="mcs-form-search-submit"
      type="submit"
    >
      Find
    </button>
  </div>
</form>
`;

exports[`component should match snapshot 3`] = `
<form
  className="mcs-form "
  onSubmit={[Function]}
>
  <div
    className="mcs-form-search "
  >
    <button
      className="mcs-form-search-prefix"
      onClick={[Function]}
      type="button"
    >
      O
    </button>
    <input
      className="mcs-form-search-input-prefix"
      name="login"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <button
      className="mcs-form-search-submit"
      type="submit"
    >
      Find
    </button>
  </div>
</form>
`;
